### PR TITLE
KAJ-670 - Mac Documentation

### PR DESCRIPTION
**Pull request by adventurist**
# Story

https://jira.rbbn.com/browse/KAJ-670

# Description

This adds a `README` with install/setup instructions and a `CHANGELOG` for KandyDistant Virtual Driver / Plugin to be used with Citrix Workspace App for MacOS.

### CODE REVIEW

**Review by pegnovi — APPROVED** [review]
LGTM.

File: mac/README.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+Citrix Workspace App for Mac 2012 (tested on 2106 in release x.x.x)
+
+Dual Core CPU with 4 GB RAM recommended (equivalent to MacBook Air 2015 and newer)
+
+## 2. Limitations
+Only 1 VDI session is permitted. In the event that a 2nd VDI session is requested, a Session Error response will be returned to the client application.
+
+## 3. Installation
+
+### 3.1 Building the Plugin and Applications
```
Comment (jfmoron) [inline mac/README.md:19]: this is a customer documentation they won't build it

File: mac/README.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+Change directories to `<KandyLib>/cpp`
+Run the `install.sh` script
+
+This will compile the plugin and applications and place them in `<KandyLib>/cpp/Default/out`
+
+### 3.2 Signing
```
Comment (jfmoron) [inline mac/README.md:25]: For signing we need to tell them what they need in the entitlements file, but we won't provide them.

File: mac/README.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+To assist in signing the product components, the following 3 plist files are provided:
+- `<KandyLib>/cpp/entitlements-dylibs.plist`
+- `<KandyLib>/cpp/src/browser/mac/entitlements-helpers.plist`
+- `<KandyLib>/cpp/src/browser/mac/entitlements-helpers.plist`
+
+### 3.3 Sign the Plugin
```
Comment (jfmoron) [inline mac/README.md:31]: let's make this generic, remove RIBBON and the paths specific to our env

File: mac/README.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+5. Verify signing:
+`codesign --verify --deep --strict --verbose=2 <output folder>/KandyDistant.dmg`
+
+## 4. Installing
+
+### 4.1 Citrix Workspace App
+It is expected that you have installed [Citrix Workspace App for Mac](https://www.citrix.com/downloads/workspace-app/mac/workspace-app-for-mac-latest.html).
+
+
+### 4.2 Installing Files
```
Comment (jfmoron) [inline mac/README.md:105]: here make the folders generic.

File: mac/README.md [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+Edit the `Modules` to add the necessary settings:
+1. Open `~/Library/Application\ Support/Citrix\ Receiver/Modules`
+2. Locate the `[ICA 3.0]` section
+3. Append the KandyDistant plugin to the `VirtualDriver` key. It should look similar to the following:
+`VirtualDriver=Thinwire3.0, TWI, SmartCard, SSPI, TUI, KandyDistant.PlugIn`
+4. Under the same section (`[ICA 3.0]`), add a key for our plugin and assign it the value of "On":
+`KandyDistant.PlugIn=On`
+5. Add a section for our plugin:
+`[KandyDistant]`
```
Comment (jfmoron) [inline mac/README.md:131]: don't we need the executable path here?

File: mac/README.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+- critical
+- error
+- warn
+- info
+- debug
+- trace
+
+### 5.3 KandyLib Configuration
+KandyLib-specific configuration can be set and modified in a config ini file which is expected to be found here:
+- ~/Library/Application Support/Kandy/config.ini
```
Comment (jfmoron) [inline mac/README.md:154]: is the config.ini relative to the app?

**Review by jfmoron — CHANGES_REQUESTED** [review]
Main comment is about this is a customer acting doc.

**Review by jfmoron — COMMENTED** [review]
You will also need to update the main page to add a link

File: mac/README.md (hunk 5 above)
Comment (adventurist) [reply mac/README.md:131]: I don't think so - at least, we never set it in our own Modules file

File: mac/README.md (hunk 1 above)
Comment (adventurist) [reply mac/README.md:19]: Changed this to "preparing the installation files" in case they need to expand an archive

File: mac/README.md (hunk 2 above)
Comment (adventurist) [reply mac/README.md:25]: we followed up on this

File: mac/README.md (hunk 3 above)
Comment (adventurist) [reply mac/README.md:31]: done

File: mac/README.md (hunk 4 above)
Comment (adventurist) [reply mac/README.md:105]: Done

File: mac/README.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+`codesign --verify --deep --strict --verbose=2 <output folder>/KandyDistant.dmg`
+
+## 4. Installing
+
+### 4.1 Citrix Workspace App
+It is expected that you have installed [Citrix Workspace App for Mac](https://www.citrix.com/downloads/workspace-app/mac/workspace-app-for-mac-latest.html).
+
+
+### 4.2 Installing Files
+*Note: This assumes that you are installing the Plugin from the usual build directory, and that you are NOT installing it from a mounted Apple Disk Image*
```
Comment (jfmoron) [inline mac/README.md:105]: this can be removed

**Review by jfmoron — CHANGES_REQUESTED** [review]

File: mac/README.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+1. Ensure that the following directories exist:
+- `~/Library/Application Support/Citrix/PlugIns`
+- `~/Library/Application Support/Kandy`
```
[the remark this answers is not in the document]
Comment (adventurist) [reply mac/README.md:108]: changed

File: mac/README.md (hunk 7 above)
Comment (adventurist) [reply mac/README.md:105]: done

**Review by jfmoron — APPROVED** [review]

**Pull request merged**